### PR TITLE
Remove MLflow integration

### PR DESCRIPTION
**Pull request by y0z**
## Motivation
- Progress #4484.

## Description of the changes
- Remove MLflow integration.
- Please merge https://github.com/optuna/optuna-integration/pull/84 before merging this PR.

### CODE REVIEW

**Comment by y0z** [comment]
In CI, `Coverage / coverage (pull_request)` failed but the cause looks unrelated to this PR.

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

File: tutorial/20_recipes/007_optuna_callback.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 ``Callback`` is called after every evaluation of ``objective``, and
 it takes :class:`~optuna.study.Study` and :class:`~optuna.trial.FrozenTrial` as arguments, and does some work.
 
-:class:`~optuna.integration.MLflowCallback` is a great example.
+`optuna_integration.MLflowCallback <https://optuna-integration.readthedocs.io/en/stable/reference/generated/optuna_integration.MLflowCallback.html>`_ is a great example.
```
Comment (HideakiImamura) [inline tutorial/20_recipes/007_optuna_callback.py:12]: How about the following?
```suggestion
`MLflowCallback <https://optuna-integration.readthedocs.io/en/stable/reference/generated/optuna_integration.MLflowCallback.html>`_ is a great example.
```

**Comment by y0z** [comment]
I applied all the comments. PTAL.

**Review by HideakiImamura — APPROVED** [review]
LGTM. BTW, could you resolve the conflict?

**Comment by y0z** [comment]
I resolved the conflict 🙌

**Pull request merged**